packages/medusa/src/api-v2/utils/middlewares/products/set-pricing-context.ts

```diff
@@ -0,0 +1,89 @@
+import { MedusaPricingContext } from "@medusajs/types"
+import { isPresent, MedusaError } from "@medusajs/utils"
+import { NextFunction } from "express"
+import { AuthenticatedMedusaRequest } from "../../../../types/routing"
+import { refetchEntities, refetchEntity } from "../../refetch-entity"
+
+export function setPricingContext() {
+  return async (req: AuthenticatedMedusaRequest, _, next: NextFunction) => {
+    // If the endpoint doesn't request prices, we can exit early
+    if (
+      !req.remoteQueryConfig.fields.some((field) =>
+        field.startsWith("variants.calculated_price")
+      )
+    ) {
+      delete req.filterableFields.region_id
+      delete req.filterableFields.currency_code
+
+      return next()
+    }
```

Comment:
Q: here, If I pass the region id and/or currency but I forget about the relation, I will still get a result but without the context filtering, it can make it hard to debug, what if we throw an invalid data here instead to say that it is not possible to filter the with the context without this relation, or alternatively we infer the context filtering if region id or currency code is present since they are only used in that context.

wdyt?